### ISSUE TITLE
Dark Mode: Less Dark Option

### ISSUE THREAD

**Issue by WildByDesign** [notepad-plus-plus/notepad-plus-plus#9982]
It is amazing to see Dark Mode in Notepad++ now. Not only does it have Dark Mode, but within that code is potential to expand upon theming colors which is great stuff.

Personally, I find the Dark Mode to be too dark and too much contrast for my use. It is also difficult to know which is the active tab. I know that it was suggested before but may have been lost within other closed issues, but I would like to suggest a Less Dark option. As was discussed in those closed issues previously, some of you had suggested less dark options similar to Twitter or Github's dimmed dark theme options.

Here is my own current build that is easier on my eyes:

![NppFluentIcons-LessDark](https://user-images.githubusercontent.com/26308319/121375229-93b88280-c90e-11eb-8404-da82084e3b35.png)

My goal was to blend the inactive tabs into the background and make the active tab pop a bit more.

The theming colors come from: https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/PowerEditor/src/NppDarkMode.cpp#L58

In my own build, I simply changed from:

```
static const Colors darkColors{
		HEXRGB(0x202020),	// background
		HEXRGB(0x282828),	// softerBackground
		HEXRGB(0x404040),	// hotBackground
		HEXRGB(0x000000),	// pureBackground
		HEXRGB(0xB00000),	// errorBackground
		HEXRGB(0xE0E0E0),	// textColor
		HEXRGB(0xC0C0C0),	// darkerTextColor
		HEXRGB(0x707070),	// disabledTextColor
		HEXRGB(0x808080),	// edgeColor
```

Changed to:

```
static const Colors darkColors{
		HEXRGB(0x202020),	// background
		HEXRGB(0x404040),	// softerBackground
		HEXRGB(0x404040),	// hotBackground
		HEXRGB(0x202020),	// pureBackground
		HEXRGB(0xB00000),	// errorBackground
		HEXRGB(0xE0E0E0),	// textColor
		HEXRGB(0xC0C0C0),	// darkerTextColor
		HEXRGB(0x808080),	// edgeColor
```

That is all that I had to change to get that little slice of heaven there. But clearly there needs to be an option within the UI that can modify these colors without having to re-compile a build. 

- Some of the suggestions were to have a Slider for the level of darkness. 
- Or several radio buttons to select from different theme options. 
- Or possibly even a color picker in which you could pick from any colors for each of these UI elements and have those colors change on the fly.

Anyway, I just wanted to get a formal issue started so that these previous ideas and suggestions can come together here.

**Comment by mere-human** [comment]
Yes, I wanted that too!

> * Or several radio buttons to select from different theme options.

I think this is could be a good starting point.

**Comment by thomassmith311** [comment]
I like the idea. Currently the preferences window is set up so that Dark Mode is a completely separate thing, but maybe it would be better to make an "Appearance" or "Theme" section, that has three radio buttons: Default Light, Default Dark, and Dark Dimmed.

**Comment by WildByDesign** [comment]
Here is my current daily build of Np++ with a concept of matching the Windows Accent Color:

![NppFluentIcons-Theme](https://user-images.githubusercontent.com/26308319/121778296-4f262480-cb64-11eb-95ec-3ad291984002.png)

That includes the recent fix by @adzm (https://github.com/notepad-plus-plus/notepad-plus-plus/commit/7e139e1ba20f3e33c0e58565aa8335ab99ffd2e4) which removes the white line under the menubar. I have manually built this to match the accent color.

The Windows Accent Color can be found here: `Computer\HKEY_CURRENT_USER\SOFTWARE\Microsoft\Windows\DWM\AccentColor`

That would be interesting to have a feature in Np++ which automatically themes the UI based on this accent color. In theory, this could be dark theme colors, purple, pink, or anything.

The theming work done thus far by @adzm is absolutely phenomenal. Your work is greatly appreciated and will be enjoyed by Np++ users for years. Your theming improvements are smooth like butter!

**Comment by adzm** [comment]
This is pretty great. I've been thinking that we should expose this in the
usual styles/themes and then allow overrides, either custom colors or using
some options like dark mode, less dark, accent color, etc.

**Comment by depascalis** [comment]
From the users perspective, there are many preferences that should be grouped under a "Appearance" section for better logical UX design. After a quick overview, these are some of the settings that more or less belong under "Appearance":

- General ( most of it )
- Editing > Caret Settings
- Editing > Line highlighting
- Editing > Smooth font
- Dark Mode
- Margins/Border/Edge
- Recent File History
- MISC > Show only file name in title bar
- Edit popup menu
- User Defined Language 
- And especially the Style configurator!

I know it's not the same for a npp devs perspective, for example Scintilla is in one place etc.

Most importantly for me however, is that the setting is exposed in the first place, and not with predefined options but rather with user defined values. I'm happy if this is configurable through some config file, however, for better UX the **color settings for UI elements should  be in the Style Configurator** (again, from a users perspective).

**Comment by MarkusBodensee** [comment]
@WildByDesign @adzm 

I am wondering, if the removal of `HEXRGB(0x707070),	// disabledTextColor` have any negative effect?

It was added with #9919 to correct text colour of disabled checkboxes, but for the moment, I do not find any visual noticeable problems. Do I miss something? The colour of `disabledTextColor` and `edgeColor` is not sooo much different.

Is maybe `edgeColor` is now treated as `disabledTextColor`? And `edgeColor` now some random value? Where is `edgeColor` used?

**Comment by MarkusBodensee** [comment]
Ok, so issue of last comment is fixed with bef4da61beabcfac8de75c3c2304b8e3e5775718 and everything fine again.
